src/Angor/Client/Pages/Founder.razor

```diff
@@ -109,63 +112,104 @@
 
         foreach (var key in keys.Keys)
         {
-            if (founderProjects.Exists(_ => _.ProjectInfo.ProjectIdentifier == key.ProjectIdentifier))
+            if (founderProjects.Any(_ => _.ProjectInfo.ProjectIdentifier == key.ProjectIdentifier))
                 continue;
 
             var indexerProject = await _IndexerService.GetProjectByIdAsync(key.ProjectIdentifier);
 
-            if (indexerProject == null)
-                break;
-
-            founderProjectsToLookup.Add(key.NostrPubKey, indexerProject);
+            if (indexerProject != null)
+            {
+                founderProjectsToLookup.Add(key.NostrPubKey, indexerProject);
+            }
         }
 
-        if (founderProjectsToLookup.Any())
+        if (!founderProjectsToLookup.Any())
         {
-            RelayService.RequestProjectCreateEventsByPubKey(e =>
+            scanningForProjects = false;
+            return;
+        }
+
+        RelayService.RequestProjectCreateEventsByPubKey(
+            e =>
+            {
+                switch (e)
                 {
-                    switch (e)
+                    case { Kind: NostrKind.Metadata }:
+                        var nostrMetadata = serializer.Deserialize<ProjectMetadata>(e.Content);
+                        var existingProject = founderProjects.FirstOrDefault(_ => _.ProjectInfo.NostrPubKey == e.Pubkey);
+
+                        if (existingProject != null)
+                        {
+                            if (existingProject.Metadata is null)
+                                existingProject.Metadata = nostrMetadata;
+                        }
+                        else
+                        {
+                            founderProjects.Add(new FounderProject
+                            {
+                                Metadata = nostrMetadata,
+                                ProjectInfo = new ProjectInfo { NostrPubKey = e.Pubkey }
+                            });
+                        }
+
+                        break;
+
+                    case { Kind: NostrKind.ApplicationSpecificData }:
+                        var projectInfo = serializer.Deserialize<ProjectInfo>(e.Content);
+                        var project = founderProjects.FirstOrDefault(_ => _.ProjectInfo.NostrPubKey == e.Pubkey);
+
+                        if (project != null)
+                        {
+                            project.ProjectInfo = projectInfo;
+                        }
+                        else
+                        {
+                            var projectIndex = founderProjectsToLookup.Keys.ToList().IndexOf(e.Pubkey) + 1;
+                            var trxId = founderProjectsToLookup[e.Pubkey].TrxId;
+
+                            founderProjects.Add(new FounderProject
+                            {
+                                ProjectInfo = projectInfo,
+                                ProjectIndex = projectIndex,
```

Comment:
what is the `ProjectIndex` is it not the index of the project in the derivation path?